v22.1/deploy-app-vercel.md

```diff
@@ -0,0 +1,177 @@
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ cd typescript/rest-nextjs-api-routes
+    ~~~
+
+## Step 3. Initialize the database
+
+1. Set the local `DATABASE_URL` environment variable to [the connection string](#connection-string) you obtained earlier from the {{ site.data.products.db }} Console:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ export DATABASE_URL="<connection-string>"
+    ~~~
+
+    By default, Prisma uses the local `DATABASE_URL` environment variable as the connection string to the database.
+
+1. Install Prisma:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    npm install prisma --save-dev
+    ~~~
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    npx prisma
+    ~~~
+
+1. Run [Prisma Migrate](https://www.prisma.io/docs/concepts/components/prisma-migrate) to initialize the database with the schema defined in `prisma/prisma.schema`.
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ npx prisma migrate dev --name init
+    ~~~
+
+    You should see the following output:
+
+    ~~~
+    Your database is now in sync with your schema.
+
+    ✔ Generated Prisma Client (3.11.0 | library) to ./node_modules/@prisma/client in 87ms
+    ~~~
+
+    This command also initializes [Prisma Client](https://www.prisma.io/docs/concepts/components/prisma-client) to communicate with your CockroachDB cluster, based on the configuration in the `prisma/schema.prisma` file.
+
+## Step 4. Deploy the application
+
+1. Save the local `DATABASE_URL` environment variable to a `.env` file in your project:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ echo "DATABASE_URL=$DATABASE_URL" >> .env
+    ~~~
+
+    Next.js loads the variables defined in `.env` to the project environment.
+
+1. (Optional) Run the app server locally to preview your site:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ npm run dev
+    ~~~
+
+1. Install the `vercel` CLI:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ npm i -g vercel
+    ~~~
+
+1. Run the `vercel` command to start deploying the app:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ vercel
+    ~~~
+
+    You will be asked to sign into your Vercel account:
+
+    ~~~
+    Vercel CLI 24.0.0
+    > > No existing credentials found. Please log in:
+    > Log in to Vercel (Use arrow keys)
+    ❯ Continue with GitHub
+      Continue with GitLab
+      Continue with Bitbucket
+      Continue with Email
+      Continue with SAML Single Sign-On
+    ~~~
+
+    When you authenticate your credentials, you should see the following message:
+
+    ~~~
+    > Success! GitHub authentication complete ...
+    ~~~
+
+    Select the default options for the remaining prompts to deploy the app to the `example-app-typescript-vercel` Vercel project:
+
+    ~~~
+    ? Set up and deploy “~/path/example-app-typescript-vercel”? [Y/n] y
```

Comment:
I think it would be nice to give users the appropriate flags instead of directing them through the interactive flow. Flags are more likely to be stable, and it's fewer steps for the user. E.g. `vercel deploy --confirm .`.